python/rikai/types/geometry.py

```diff
@@ -392,6 +394,31 @@ def iou(
             return iou_arr[0]
         return iou_arr
 
+    def with_label(
+        self, text: str, color: str = get_option(CONF_RIKAI_VIZ_COLOR)
+    ) -> "rikai.viz.Draw":
+        from rikai.viz import Draw, Text
+
+        """return a `rikai.viz.Draw`, which contains a label,
+        used as a convenient tool to give a box a label
+        in visualization.
+        Parameters
+        ----------
+        text: str
+            The text content of that label
+        color: str
+            The color of the text,
+            will have a default value if not given.
+        Returns
+        -------
+        box_with_label
+            rikai.viz.Draw
+        """
+        # TODO add font size
+        return Draw(
+            [self, Text(text, (int(self.xmin), int(self.ymin)), color)]
```

Comment:
as I suggested in #632 , we should shift the (xmin, ymin) like (xmin, ymin-10), and 10 should be the default height (or a bit bigger) of the font size